openapi_python_client/templates/property_templates/date_property.pyi

```diff
@@ -9,12 +9,12 @@ if _{{ property.python_name }} is not None:
 {% endif %}
 {% endmacro %}
 
-{% macro transform(property, source, destination, declare_type=True) %}
+{% macro transform(property, source, destination, declare_type=True, query_parameter=False) %}
 {% if property.required %}
```

Comment:
I think we also need this? I could be confusing myself now 🥸

```suggestion
{% if property.required and (not query_parameter or not property.nullable) %}
```

Reply:
Wouldn't we be safe in that case by the `{% if property.nullable %}if {{ source }} else None {%endif%}` on L14?

Reply:
Oh I see, I was thinking that in endpoint_macros we were only ignoring params that were UNSET, not also `None`.

Actually though I think I just spotted another issue but I can't comment on it. Line 25 in endpoint_macros will include all params if they are required, even if they are `None`. If we truly want to treat nullable and optional as the same thing we'll need to adjust that `if` and the one on line 35.